web3/eth.py

```diff
@@ -49,6 +51,147 @@
+
+# 300000 blocks
+GET_LOGS_BATCH_SIZE: int = 300000
+
+GET_LOGS_MAX_RETRIES: int = 5
+
+# cache timeout in ms
+CACHE_TIMEOUT: int = 60000
+
+# 50 blocks
+CONFIRMATION_BLOCKS: int = 50
+
+logger = logging.getLogger(__name__)
+
+
+class EthGetLogsCachedValue:
+
+    def __init__(self,
+                 address: str,
+                 topics: List[str],
+                 from_block: int,
+                 to_block: int,
+                 logs: List[dict]):
+        self.address = address
+        self.topics = topics
+        self.from_block = from_block
+        self.to_block = to_block
+        self.logs = logs
+
+    def get_logs(self, from_block: int, to_block: int):
+        logs = list()
+        for log in self.logs:
+            # this is needed since rskj sometimes provide events without blockNumber
+            # (that's something that sometimes happens but not always, seems to be a bug on the core)
+            log_block_number: int = log['blockNumber'] if "blockNumber" in log else None
+            if log_block_number and from_block <= log_block_number <= to_block:
+                logs.append(log)
+        logger.debug(f"Getting logs from cached value from block {from_block} to block {to_block} -> logs {logs}")
+        return logs
+
+
+class EthGetLogsCache:
+
+    logs: Dict[int, EthGetLogsCachedValue]
+
+    def __init__(self, web3):
+        self.logs = {}
+        self.web3 = web3
+
+    @staticmethod
+    def get_next_batch_end(current_batch_start: int,
+                           final_block_number: int,
+                           batch_size: int = GET_LOGS_BATCH_SIZE) -> int:
+        current_batch_end = current_batch_start + batch_size
+        return min(current_batch_end, final_block_number)
+
+    def get_logs(self, filter_params):
+        from_block: int = filter_params["fromBlock"]
+        to_block: int = filter_params["toBlock"]
+        batch_size: int = filter_params["batchSize"] if "batchSize" in filter_params else GET_LOGS_BATCH_SIZE
+        retries: int = filter_params["retries"] if "retries" in filter_params else GET_LOGS_MAX_RETRIES
+        if "batchSize" in filter_params:
+            del filter_params["batchSize"]
+        if "retries" in filter_params:
+            del filter_params["retries"]
+        current_batch_start: int = int(from_block)
+        current_batch_end: int = EthGetLogsCache.get_next_batch_end(current_batch_start, to_block, batch_size)
+        logs = list()
+        while current_batch_start < current_batch_end:
+            filter_params["fromBlock"] = current_batch_start
+            filter_params["toBlock"] = current_batch_end
+            log_list = None
+            for currentTry in range(1, (retries + 1)):
+                try:
+                    logger.debug(f'Trying to get logs with params {filter_params} '
+                                 f'from block {current_batch_start} to block {current_batch_end}. '
+                                 f'(attempt {currentTry})')
```

Comment:
```suggestion
            for currentTry in range(retries):
                try:
                    logger.debug(f'Trying to get logs with params {filter_params} '
                                 f'from block {current_batch_start} to block {current_batch_end}. '
                                 f'(attempt {currentTry + 1})')
```

for me this is more readable, but not a blocker